### PR TITLE
NEX-15: switch from jsonapi to graphql

### PR DESCRIPTION
**Pull request by vermario**
# Dismiss JSONAPI, switch to GraphQL for content data fetching, add graphql-codegen for type generation

## Drupal side changes:
* Add the `graphql` and `graphql_compose` modules
* Add existing entities to the schema
* Add view to display articles with a GraphQL display, used in client and server side listings
* Remove jsonapi extra modules from composer and recipe, remove patches
* update permissions

## Next.js changes:

* *Major refactoring* of data fetching
* add graphql-codegen to generate types out of the drupal GraphQL schema and the queries/fragments defined in the frontend
* add needed queries to:
   * Get menus
   * get static paths
   * get articles
   * get all paragraphs when rendering `page` and `frontpage` nodes
   * get media
* better structure of the components folder
* removal of most of Zod schemas (still there for webform submissions, and env variables)
* refactor of listings (client and server side)
* refactor of statically generated /all-articles page 
* handling of types for queries and components using fragments and inferred types

## To test

To get to a working site, it should be enough to:
1. check out the repository
2. run `./setup.sh`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,6 +150,14 @@ The frontend uses [TypeScript](https://www.typescriptlang.org) to provide type s
 
 TypeScript is setup quite loosely by default to minimise friction and make it accessible to developers who are not familiar with it. It is recommended to increase type safety by enabling some of the disabled rules in `next/eslint.json`.
 
+#### Working with GraphQL and TypeScript
+
+The project uses GraphQL to fetch data from the backend. The queries are defined in the `next/lib/queries` directory. The queries are typed using the `graphql-codegen` package, which generates TypeScript types from the queries. The types are then used to type the data fetched from the backend.
```
Comment (joshua-scott) [inline README.md:155]:
```suggestion
The project uses GraphQL to fetch data from the backend. The queries are defined in the `next/lib/graphql/queries` directory. The queries are typed using the `graphql-codegen` package, which generates TypeScript types from the queries. The types are then used to type the data fetched from the backend.
```

File: README.md [hunk 2]
```diff
@@ -150,6 +150,14 @@ The frontend uses [TypeScript](https://www.typescriptlang.org) to provide type s
...
+
+When adding or modifying queries and fragments, you will need to run `lando npm run graphql-codegen` to generate the corresponding types. The command will keep checking the files for changes. 
+
+When there are changes on the GraphQL server schema itself, you need to stop and start the command again to fetch the new schema definition. Also, you might need to `lando drush cr` to clear the Drupal cache.
+
 #### Data fetching with TypeScript and Zod
```
Comment (joshua-scott) [inline README.md:161]: We can delete this zod section now 🎉

**Review by joshua-scott — APPROVED** [review]
Ran the setup script locally and everything works perfectly. We've also been using this approach in our project and it's been a great improvement over jsonapi 👍

File: README.md (hunk 2 above)
Comment (vermario) [reply README.md]: Deleted!

**Pull request merged**